src/domain/Rank.js

```diff
@@ -0,0 +1,71 @@
+import { LOTTO_RANK } from '../constants/lottoStandard.js';
+
+const WINNING_COUNT = Object.freeze({
+  FIRST_RANK: 6,
+  SECOND_THIRD_RANK: 5,
+  FOURTH_RANK: 4,
+  FIFTH_RANK: 3,
+});
+
+class Rank {
+  #rank;
+
+  constructor(winningNumbers, bonusNumber, userNumbers) {
+    const { matchNumbers, matchBonusNumber } = this.#matchWinning(
+      winningNumbers,
+      bonusNumber,
+      userNumbers,
+    );
+    this.#generateRank(matchNumbers, matchBonusNumber);
+  }
+
+  #matchWinning(winningNumbers, bonusNumber, userNumbers) {
+    const matchNumbers = winningNumbers.filter((number) => userNumbers.includes(number));
+    const matchBonusNumber = userNumbers.includes(bonusNumber);
+
+    return { matchNumbers, matchBonusNumber };
+  }
+
+  #generateRank(matchNumbers, matchBonusNumber) {
+    this.#checkFirstRank(matchNumbers.length);
+    this.#checkSecondThridRank(matchNumbers.length, matchBonusNumber);
+    this.#checkRestRank(matchNumbers.length);
+  }
+
+  #checkFirstRank(length) {
+    if (length === WINNING_COUNT.FIRST_RANK) {
+      this.#rank = LOTTO_RANK.FIRST_RANK;
+    }
+  }
+
+  #checkSecondThridRank(length, matchBonusNumber) {
+    if (length === WINNING_COUNT.SECOND_THIRD_RANK) {
+      if (matchBonusNumber) {
+        this.#rank = LOTTO_RANK.SECOND_RANK;
+        return;
+      }
+
+      this.#rank = LOTTO_RANK.THIRD_RANK;
+    }
+  }
+
+  #checkRestRank(length) {
+    if (length === WINNING_COUNT.FOURTH_RANK) {
+      this.#rank = LOTTO_RANK.FOURTH_RANK;
+    }
+
+    if (length === WINNING_COUNT.FIFTH_RANK) {
+      this.#rank = LOTTO_RANK.FIFTH_RANK;
+    }
+
+    if (length < WINNING_COUNT.FIFTH_RANK) {
+      this.#rank = LOTTO_RANK.SIXTH_RANK;
+    }
+  }
```

Comment:
특정 조건을 나열할 때는 `switch` 문 사용을 추천드립니다!
유지보수성, 확장성, 가독성 측면에서 if문을 나열하는 것보다 좋을 것 같아요:)

```suggestion
  #checkRestRank(length) {
    switch (length) {
      case WINNING_COUNT.FOURTH_RANK:
        this.#rank = LOTTO_RANK.FOURTH_RANK;
        break;
      case WINNING_COUNT.FIFTH_RANK:
        this.#rank = LOTTO_RANK.FIFTH_RANK;
        break;
    ...
    }
  }
```